llvm/lib/Target/NVPTX/NVPTXAsmPrinter.cpp

```diff
@@ -506,24 +507,15 @@ void NVPTXAsmPrinter::emitKernelFunctionDirectives(const Function &F,
   // If the NVVM IR has some of reqntid* specified, then output
   // the reqntid directive, and set the unspecified ones to 1.
   // If none of Reqntid* is specified, don't output reqntid directive.
-  std::optional<unsigned> Reqntidx = getReqNTIDx(F);
-  std::optional<unsigned> Reqntidy = getReqNTIDy(F);
-  std::optional<unsigned> Reqntidz = getReqNTIDz(F);
+  const auto ReqNTID = getReqNTID(F);
+  if (!ReqNTID.empty())
+    O << formatv(".reqntid {0:$[, ]}\n",
+                 make_range(ReqNTID.begin(), ReqNTID.end()));
```

Comment:
`getReqNTID()` already returns a small vector which should be accepted as a range input. Do we really need an explicit range creation here?

Reply:
I agree it seems like it should not be necessary, but this was the only way I could get it to compile. If you know a cleaner/more idiomatic way to do this that would be great.

Reply:
Looks like `make_range()` is all we have now. That's what formatv() tests use, too:
https://github.com/llvm/llvm-project/blob/f62f13d5db212b4bebe6fc143fb9827703e88dfd/llvm/unittests/Support/FormatVariadicTest.cpp#L607-L612